hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/table/HoodieFlinkCopyOnWriteTable.java

```diff
@@ -371,7 +371,7 @@ protected Iterator<List<WriteStatus>> handleUpdateInternal(HoodieMergeHandle<?,
       throw new HoodieUpsertException(
           "Error in finding the old file path at commit " + instantTime + " for fileId: " + fileId);
     } else {
-      HoodieMergeHelper.newInstance().runMerge(this, upsertHandle);
+      HoodieMergeHelper.newInstance().runMerge(this, upsertHandle, Option.empty(), new Object[0]);
```

Comment:
We can avoid all such changes if the partition fields and values can be deduced from base path and file path.

Reply:
The problem is that we have been using spark to get those values and MergeHelper is supposed to be engine agnostic